lisa/sut_orchestrator/azure/features.py

```diff
@@ -783,3 +788,128 @@ def get_azure_disk_type(disk_type: schema.DiskType) -> str:
     schema.DiskType.StandardHDDLRS: "Standard_LRS",
     schema.DiskType.StandardSSDLRS: "StandardSSD_LRS",
 }
+
+
+class Resize(AzureFeatureMixin, features.Resize):
+    def resize(
+        self, resize_action: ResizeAction = ResizeAction.IncreaseCoreCount
+    ) -> schema.NodeSpace:
+        platform: AzurePlatform = self._platform  # type: ignore
+        compute_client = get_compute_client(platform)
+        node_context = get_node_context(self._node)
+        new_vm_size_info = self._select_vm_size(resize_action)
+
+        # Creating parameter for VM Operations API call
+        hardware_profile = HardwareProfile(vm_size=new_vm_size_info.vm_size)
+        vm_update = VirtualMachineUpdate(hardware_profile=hardware_profile)
+
+        # Resizing with new Vm Size
+        lro_poller = compute_client.virtual_machines.begin_update(
+            resource_group_name=node_context.resource_group_name,
+            vm_name=node_context.vm_name,
+            parameters=vm_update,
+        )
+
+        # Waiting for the Long Running Operation to finish
+        timeout = 60 * 5
+        timer = create_timer()
+        while timeout > timer.elapsed(False):
```

Comment:
It's a nice implementation. Please update the `wait_operation` with the timeout check, and check the cancellation too.

Reply:
could you expand on this plz. I don't really understand

Reply:
There is a method called `wait_operation` in `azure/common.py`. It will wait any Azure operation complete. Please add your loop logic into that method, and use the method here.